src/main/java/org/mockito/Mockito.java

```diff
@@ -2333,6 +2351,30 @@ public static VerificationMode only() {
     }
 
     /**
+     * Allows counting of number of invocations. E.g:
+     * <pre class="code"><code class="java">
+     *   AtomicInteger counter = new AtomicInteger();
+     *   verify(mock, countInvocations(counter)).someMethod("some arg");
+     *   verify(otherMock, times(counter)).someMethodThatShouldAlsoHaveBeenCalled();
+     * </code></pre>
+     *
+     * or by directly asserting the counter:
+     *
+     * <pre class="code"><code class="java">
+     *   AtomicInteger counter = new AtomicInteger();
+     *   verify(mock, countInvocations(counter)).someMethod("some arg");
+     *   assertEquals(2, counter.get());
+     * </code></pre>
```

Comment:
I'd rather see something like that

```java
InvocationStats someMethodStats = AdditionalVerifications.invocationStats();
verify(mock, stats).someMethod("some arg");
assertEquals(2, someMethodStats.count());
```

Note I used the word stat to allow possible other metrics (**if relevant**). Also this should be annotated with `@Incubating`